src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -252,7 +254,8 @@ public static string SelectProductNameForDirectory(Platform.XDG_Type dirpath)
             string envHome = System.Environment.GetEnvironmentVariable(CommonEnvVariableNames.Home);
             if (envHome == null)
             {
-                envHome = GetTemporaryDirectory();
+                s_tempHomeDir ??= GetTemporaryDirectory();
+                envHome = s_tempHomeDir;
```

Comment:
You assume that HOME could be defined on the fly (and preserve current behavior) but we could already put some files in previous folder :-(

Reply:
Sorry that I missed this comment. I think it would be very unlikely that the `HOME` env var changes it value after it's declared, so I'm not too worry about this.